src/api/consumer/lib/import.ts

```diff
@@ -165,39 +165,45 @@ function warnForPackageDependencies({ dependencies, consumer, installNpmPackages
     )
   );
 
-  // eslint-disable-next-line
-  dependencies.forEach(dep => {
-    if (!dep.packageDependencies || R.isEmpty(dep.packageDependencies)) return null;
-
-    R.forEachObjIndexed((packageDepVersion, packageDepName) => {
-      const packageDep = { [packageDepName]: packageDepVersion };
-      const compatibleWithPackgeJson = compatibleWith(packageDep, packageJsonDependencies);
-      const compatibleWithNodeModules = compatibleWith(packageDep, nodeModules);
-
-      if (!compatibleWithPackgeJson && !compatibleWithNodeModules && !R.contains(packageDep, warnings.notInBoth)) {
-        // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-        warnings.notInBoth.push(packageDep);
-      }
+  // since librarian does not install node_modules on the hd,
+  // transient dependencies will be neither in the root package.json
+  // nor in the (non-existent) node_modules
+  // TODO: a better check would be to check the dependency package.json files
+  if (consumer.config.packageManager !== 'librarian') {
```

Comment:
why not change it to be
```
if (consumer.config.packageManager === 'librarian') {
// Remove duplicates warnings for missing packages
  warnings.notInBoth = R.uniq(warnings.notInBoth);

  return Promise.resolve(warnings);
}
Always prefer the shorter code in the if
```

Reply:
Well, this:
```
// Remove duplicates warnings for missing packages
  warnings.notInBoth = R.uniq(warnings.notInBoth);

  return Promise.resolve(warnings);
```
Happens for all package managers. So if we swap the if around, we'll have to duplicate these lines. Or am I misunderstanding?

Reply:
I don't think you need the `warnings.notInBoth = R.uniq(warnings.notInBoth);` in case of librarian since you don't have the `warnings.notInBoth.push(packageDep);` you shouldn't have duplicates anyway. 
so you can put in the if just `return Promise.resolve(warnings);`
(I'm not sure you even need the `warnings` i mean it will always object with empty arrays, but keep it so you won't need to change the calling function)